lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.parquet.PrimitiveField;
+import io.trino.parquet.reader.flat.BooleanFlatColumnReader;
+import io.trino.parquet.reader.flat.ByteFlatColumnReader;
+import io.trino.parquet.reader.flat.IntFlatColumnReader;
+import io.trino.parquet.reader.flat.LongFlatColumnReader;
+import io.trino.parquet.reader.flat.ShortFlatColumnReader;
+import io.trino.spi.TrinoException;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.LogicalTypeAnnotation.DecimalLogicalTypeAnnotation;
+import org.apache.parquet.schema.LogicalTypeAnnotation.TimeLogicalTypeAnnotation;
+import org.apache.parquet.schema.LogicalTypeAnnotation.TimestampLogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName;
+import org.joda.time.DateTimeZone;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static io.trino.parquet.ParquetTypeUtils.createDecimalType;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static java.lang.Boolean.FALSE;
+import static java.lang.Boolean.TRUE;
+import static java.util.Objects.requireNonNull;
+
+public final class ColumnReaderFactory
+{
+    private ColumnReaderFactory() {}
+
+    public static ColumnReader create(PrimitiveField field, DateTimeZone timeZone)
```

Comment:
This is only used in tests. Move it there, instead.

Reply:
We use it in `ParquetReader#initializeColumnReaders` in this commit